### PR TITLE
add filter offices users consumer

### PR DESCRIPTION
**Pull request by borovHokage**

### CODE REVIEW

File: src/OfficeService.Data/OfficeUserRepository.cs [hunk 1]
```diff
@@ -43,6 +43,14 @@ public async Task<List<DbOfficeUser>> GetAsync(List<Guid> usersIds)
       return await users.ToListAsync();
     }
 
+    public async Task<List<DbOfficeUser>> GetOfficeAsync(List<Guid> officesIds)
```
Comment (MariyaSin) [inline src/OfficeService.Data/OfficeUserRepository.cs]: use OfficeRepository to get Offices

Comment (borovHokage) [reply src/OfficeService.Data/OfficeUserRepository.cs]: fixed

File: src/OfficeService.Broker/Consumers/FilterOfficesUsersConsumer.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      _repository = repository;
+    }
+
+    public async Task Consume(ConsumeContext<IFilterOfficesRequest> context)
+    {
+      List<OfficeFilteredData> officesFilteredData = await GetOfficesDataAsync(context.Message);
+
+      await context.RespondAsync<IOperationResult<IFilterOfficesResponse>>(
+        OperationResultWrapper.CreateResponse((_) => IFilterOfficesResponse.CreateObj(officesFilteredData), context));
+
```
Comment (MariyaSin) [inline src/OfficeService.Broker/Consumers/FilterOfficesUsersConsumer.cs]: remove empty line

File: src/OfficeService.Broker/Consumers/FilterOfficesUsersConsumer.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    private async Task<List<OfficeFilteredData>> GetOfficesDataAsync(IFilterOfficesRequest request)
+    {
+      List<DbOffice> offices = await _repository.GetAsync(request.OfficesIds);
+
+      return offices.Select(o =>
+          new OfficeFilteredData(
+            o.Id,
+            o.Name,
+            o.Users.Where(u => u.IsActive).Select(u => u.UserId).ToList()))
```
Comment (MariyaSin) [inline src/OfficeService.Broker/Consumers/FilterOfficesUsersConsumer.cs]: remove Users.Where(u => u.IsActive) to repository method

**Review by MariyaSin — APPROVED** [review]

File: src/OfficeService.Broker/Consumers/FilterOfficesUsersConsumer.cs (hunk 2 above)
Comment (borovHokage) [reply src/OfficeService.Broker/Consumers/FilterOfficesUsersConsumer.cs]: done

File: src/OfficeService.Broker/Consumers/FilterOfficesUsersConsumer.cs (hunk 3 above)
Comment (borovHokage) [reply src/OfficeService.Broker/Consumers/FilterOfficesUsersConsumer.cs]: done

**Review by ndoriomedov — APPROVED** [review]

**Pull request merged**